onyo/lib/onyo.py

```diff
@@ -563,49 +564,73 @@ def validate_anchors(self) -> bool:
+                       exclude: Iterable[Path] | Path | None = None,
+                       depth: int = 0,
+                       types: List[Literal['assets', 'directories', 'templates']] | None = None
+                       ) -> List[Path]:
+        r"""Select all items in the repository that are relative to the given
         `subtrees` descending at most `depth` directories.
 
         Parameters
         ----------
         include
-          Paths to look for assets under. Defaults to the root of the inventory.
+          Paths to look for items under. Defaults to the root of the inventory.
         exclude
-          Paths to exclude, meaning that assets underneath any of these are not
+          Paths to exclude, meaning that items underneath any of these are not
           being returned. Defaults to `None`.
         depth
           Number of levels to descend into. Must be greater equal 0.
           If 0, descend recursively without limit. Defaults to 0.
+        types
+          List of types of inventory items to consider. Valid types are
+          'assets', 'directories' and 'templates'. Defaults to ['assets'].
 
         Returns
         -------
           list of Path
-            Paths to all matching assets in the repository.
+            Paths to all matching items in the repository.
         """
+        # TODO: There's a lot of potential optimization. For example: When we have types=['templates'],
+        #       there's no need to iterate over alö files in git. Conversely, w/o 'templates' there's nothing of
+        #       interest underneath `.onyo/` at all.
+        if types is None:
+            types = ['assets']
+        if include is None:
+            include = [self.git.root]
         if depth < 0:
             raise ValueError(f"depth must be greater or equal 0, but is '{depth}'")
         # Note: The if-else here doesn't change result, but utilizes `GitRepo`'s cache:
         files = self.git.get_subtrees(include) if include else self.git.files
         if depth:
-            roots = include if include else [self.git.root]
             files = [f
                      for f in files
-                     for r in roots
-                     if r in f.parents and len(f.parents) - len(r.parents) <= depth]
+                     for root in include
+                     if root in f.parents and len(f.parents) - len(root.parents) <= depth]
 
         if exclude:
             exclude = [exclude] if isinstance(exclude, Path) else exclude
             files = [f for f in files if all(f != p and p not in f.parents for p in exclude)]
 
-        # This only checks for `is_inventory_path`, since we already
-        # know it's a committed file:
-        return [f for f in files if self.is_inventory_path(f)] + \
-               [f.parent for f in files if f.name == self.ASSET_DIR_FILE_NAME]
+        paths = []
+        if 'assets' in types:
+            # This only checks for `is_inventory_path`, since we already
```

Comment:
I don't find this comment helpful. Nothing in this function would lead to believe that we're modifying the repo. So a statement that files are committed is superfluous, and could just as relevantly precede the majority of lines in the codebase.

Reply:
Well, may be I should rephrase instead. The intuitive thing to check here, seems to be `is_asset_path` rather than `is_inventory_path`. The point here is, that we don't check an arbitrary, user-provided path, but one that we already know to be a committed file. Hence, what remains to check is cheaper.